src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -607,15 +606,28 @@ public Snapshot makeSnapshotFromSnapshotRequest(SnapshotRequestModel snapshotReq
         String snapshotQuery = queryModel.getQuery();
         Query query = Query.parse(snapshotQuery);
         String datasetName = query.getDatasetName();
-        createSnapshotTablesFromDatasetAsset(datasetName, assetName, snapshotSource, snapshot);
+        Dataset queryDataset = datasetService.retrieveByName(datasetName);
+        AssetSpecification queryAssetSpecification =
+            getAssetByNameFromDataset(queryDataset, assetName);
+        snapshotSource.assetSpecification(queryAssetSpecification);
+        conjureSnapshotTablesFromAsset(snapshot, snapshotSource);
       }
       case BYROWID -> {
         SnapshotRequestRowIdModel requestRowIdModel = requestContents.getRowIdSpec();
         conjureSnapshotTablesFromRowIds(requestRowIdModel, snapshot, snapshotSource);
       }
       case BYREQUESTID -> {
-        String datasetName = snapshotRequestModel.getContents().get(0).getDatasetName();
-        createSnapshotTablesFromDatasetAsset(datasetName, ASSET_NAME, snapshotSource, snapshot);
+        String datasetName = requestContents.getDatasetName();
+        Dataset queryDataset = datasetService.retrieveByName(datasetName);
```

Comment:
Same question - is this different from `dataset` above?

Reply:
Yes, this appears to be the same dataset. I'll remove this chunk of code.

Reply:
And I'll switch to grabbing the dataset via id instead of name.